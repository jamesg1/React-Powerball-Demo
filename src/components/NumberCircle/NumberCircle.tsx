import React from 'react';
import { Circle, Label } from './Styled';

export interface Props {
  number?: number;
  powerballNumber?: boolean;
}

const powerBallLabel = 'PB';

const NumberCircle: React.FC<Props> = ({ number, powerballNumber }: Props) => {
  const label = !number && powerballNumber ? powerBallLabel : number;
  return (
    <Circle number={number} powerballNumber={powerballNumber}>
      <Label>{label}</Label>
    </Circle>
  );
};

export default NumberCircle;
