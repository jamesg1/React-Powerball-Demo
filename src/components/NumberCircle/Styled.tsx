/* eslint-disable @typescript-eslint/no-explicit-any */
import styled, { css } from 'styled-components';
import { Props } from './NumberCircle';

const darkGrey = '#474747';
const grey = '#dddddd';
const silver = '#879bab';
const blue = '#3450a7';
const white = '#ffffff';
const fontFamily = 'Muli, sans-serif';

const getBackground = (number?: number, powerballNumber?: boolean): string => {
  if (number && powerballNumber) return silver;
  if (number) return blue;
  return white;
};

const getStyling = ({ number, powerballNumber }: Props): any => {
  const backgroundColor = getBackground(number, powerballNumber);
  const textColor = !number ? darkGrey : white;
  const borderColor = !number ? grey : backgroundColor;
  const boxShadow = !number ? `box-shadow: inset 0px 2px 2px ${grey}` : '';

  return css`
    background-color: ${backgroundColor};
    border: 2px solid ${borderColor};
    color: ${textColor};
    ${boxShadow};

    &:first-of-type {
      box-shadow: none;
      ${({ number }: Props): string => (!number ? `border: 2px solid ${blue}` : '')};
    }
  `;
};

const Circle = styled.div<Props>`
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: ${fontFamily};
  font-size: 12px;
  font-weight: bold;
  margin-right: 3px;
  width: 22px;
  height: 22px;
  transition: all 400ms ease-in;

  @media (min-width: 321px) {
    width: 25px;
    height: 25px;
    margin-right: 5px;
  }

  border-radius: 50%;
  ${(props: Props): any => getStyling(props)};
`;

const Label = styled.span``;

export { Circle, Label };
