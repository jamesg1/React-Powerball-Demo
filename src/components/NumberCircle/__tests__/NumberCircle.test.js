import React from 'react';
import NumberCircle from '../NumberCircle';
import renderWithTheme from 'styling/withTheme';
import 'jest-styled-components';

describe('<NumberCircle />', () => {
  test('it renders primary number circle', () => {
    const numbers = [22, 31];
    const tree = renderWithTheme(<NumberCircle number={numbers[1]} />);
    const testInstance = tree.root;
    expect(testInstance.findByType('span').props.children).toBe(31);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('it renders powerball number circle', () => {
    const numbers = [19];
    const tree = renderWithTheme(<NumberCircle number={numbers[0]} isPowerball={true} />);
    const testInstance = tree.root;
    expect(testInstance.findByType('span').props.children).toBe(19);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('it renders secondary number circle with powerball label', () => {
    const tree = renderWithTheme(<NumberCircle isPowerball={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
