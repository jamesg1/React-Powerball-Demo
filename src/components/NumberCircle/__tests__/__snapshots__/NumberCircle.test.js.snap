// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberCircle /> it renders powerball number circle 1`] = `
<div
  className=""
  data-cy="numberCircle__powerball"
>
  <span
    className="Styled__Label-c3m4bn-1"
  >
    19
  </span>
</div>
`;

exports[`<NumberCircle /> it renders primary number circle 1`] = `
<div
  className=""
  data-cy="numberCircle__primary"
>
  <span
    className="Styled__Label-c3m4bn-1"
  >
    31
  </span>
</div>
`;

exports[`<NumberCircle /> it renders secondary number circle with powerball label 1`] = `
<div
  className=""
  data-cy="numberCircle__powerball"
>
  <span
    className="Styled__Label-c3m4bn-1"
  >
    PB
  </span>
</div>
`;
